broadinstitute/pearl
# AR-72 Preregistration workflow

This enables preregistration workflow for a study within a portal.  In the process, this ports over a lot of surveyJS functionality from the demo project, with slight upgrades to typescript it and make the data sent to the server more robustly typed.  I'll put a meeting on the calendar for either Friday or Monday of next week to talk through some possible designs for how we need to represent participant data beyond just storing it as survey snapshots.

TO TEST:
1) nuke/reset database
2) run `./scripts/populate_portal.sh ourhealth`
3) go to `http://sandbox.ourhealth.localhost:3001/`
4) click the "join" button
5) see a prereg survey
6) complete it
7) confirm you are taken to a registration page (which will error if you try to submit it)
